### PR TITLE
ref(snuba): Add instrumentation to group queryset

### PR DESCRIPTION
**Pull request by RyanSkonnord**
The database query that underlies this Django query is a target for
optimization. Add a span that will highlight when it is called and
capture the result size.

This will be useful in correlating the query's time cost to cases where
the result set is too large to use anyway. If the time cost is large in
such cases, it would justify making a separate 'count' query.

### CODE REVIEW

**Comment by RyanSkonnord** [comment]
Here is what the change looks like in the Discover view.

Screen Shot 2020-04-20 at 7 01 42 PM

Previously the child `db` entry would be another one in the flat list of `db` entries.

File: src/sentry/search/snuba/executors.py [hunk 1]
```diff
@@ -325,9 +326,14 @@ def query(
         # to something that we can send down to Snuba in a `group_id IN (...)`
         # clause.
         max_candidates = options.get("snuba.search.max-pre-snuba-candidates")
-        too_many_candidates = False
-        group_ids = list(group_queryset.values_list("id", flat=True)[: max_candidates + 1])
+
+        with sentry_sdk.start_span(op="snuba_group_query") as span:
```
Comment (RyanSkonnord) [inline src/sentry/search/snuba/executors.py:330]: The `op` name could be changed. I'm mildly concerned that the relationship to the existing `start_snuba_query` span might be a little unclear.

To explain, this "Snuba group query" is not the Snuba query proper, but a query that the Snuba executor is making against the Postgres DB in order to set up the actual search query.

**Comment by jusigler** [comment]
This looks good to me.  (I'll let another engineer 👍🏻on the PR)

**Review by markstory — APPROVED** [review]
Looks reasonable to me.

**Review by leedongwei — APPROVED** [review]

**Pull request merged**